### PR TITLE
chores: add editorconfig and dev dependencies

### PR DESCRIPTION
**Pull request by SethFalco**
Just a small chore.

While contributing to [plugin.video.invidious](https://github.com/lekma/plugin.video.invidious/), I've been reviewing this repo. I've also worked with it a bit, mostly to experiment with UI changes, like using the [details view for the select dialog](https://xbmc.github.io/docs.kodi.tv/master/kodi-base/d6/de8/group__python___dialog.html#gad00396a6c918a9406f14cd0a5fdeaa4e) to display more info, not sure if it's worthwhile, though.

While reviewing the [JellyCon](https://github.com/jellyfin/jellycon/blob/master/requirements-dev.txt) (Jellyfin) add-on for Kodi, and noticed they had a dependency called Kodistub.

> Kodi stubs are Python files that can help you develop addons for [Kodi (XBMC)](http://kodi.tv/) Media Center. Use them in your favorite IDE to enable autocompletion and view docstrings for Kodi Python API functions, classes and methods. Kodistubs also include [PEP-484](https://www.python.org/dev/peps/pep-0484/#suggested-syntax-for-python-2-7-and-straddling-code) type annotations for all functions and methods.
> 
> — https://pypi.org/project/Kodistubs/20.0.0/

Basically, it's a pip package with just types/docs in it for the most part. No real functionality.

It's made reading code and contributing _so much easier_. Thoughts on including the dev dependency here? No need to include it in the Invidious add-on repo, since it never directly interacts with xbmc API's.

---

Also adds an `.editorconfig`, similar to:
* https://github.com/lekma/plugin.video.invidious/pull/91

### CODE REVIEW

**Comment by lekma** [comment]
I get the value of KodiStubs (even though I was not aware of its existence until now and wish I was when I started messing with kodi) but I don't understand the need to add it as a requirement to this lib.
Does your editor parse requirement files and automatically install stuff?
(AFAIK kodi  doesn't use `pip` nor `setuptools` to install those addons, right?)

To be clear, I'm not against it (I just would like to understand how it works).

**Comment by lekma** [comment]
also was not aware of Jellyfin...

**Comment by SethFalco** [comment]
1. There is 100% no need for the repo to have it. Someone who is aware of it can just install `Kodistubs` themselves and get the benefits. The motivation was just for the uninitiated to get a hint that it exists. (It's more like intuitively placed documentation.)
2. My editor doesn't require this for it to work. Back when I installed it from JellyCon, I got the types/docs for all repos.
3. Afaik, nah Kodi won't do anything with the requirements file. Though if nothing is being excluded, then it could be nice to define a script to create a zip just so bloat isn't packaged.

If you don't want to merge it, I don't mind! Though I definitely think a mention to Kodistubs somewhere would be worthwhile, but no pressure. :+1: 

* Either `requirements-dev.txt` - as I think this is intuitive to Python developers.
* In the README under a **Contributing** section.
* In a `CONTRIBUTING.md` file.

**Comment by SethFalco** [comment]
> also was not aware of Jellyfin...

If you're into self-hosting your media, definitely worth checking out. :sunglasses: 
If it helps, it's a fork of Emby if you're familiar with that.

Otherwise, I'll leave you to the website/docs. :+1:

**Comment by lekma** [comment]
all right, thanks for the info
